jax/experimental/host_callback.py

```diff
@@ -278,16 +440,15 @@ def id_tap(tap_func, arg, *, result=None, tap_with_device=False, **kwargs):
 
   For more details see the
   `module documentation
-  <https://jax.readthedocs.io/en/latest/jax.experimental.host_callback.html>`_.
+  <jax.experimental.host_callback.html>`_.
   """
   if kwargs:
-    warnings.warn(
-        "Support for **kwargs in ``id_tap`` is deprecated and will be removed "
-        "in the future. Instead, pre-apply keyword arguments, either by using "
-        "a closure or by passing ``functools.partial(tap_func, **kwargs)`` "
-        "instead.",
-        FutureWarning, stacklevel=2)
-    tap_func = functools.partial(tap_func, **kwargs)
+    msg = (
+        "Support for **kwargs in ``id_tap`` has been removed. Instead, "
+        "pre-apply keyword arguments, either by using a closure or by passing "
+        "``functools.partial(tap_func, **kwargs)``.")
+    raise ValueError(msg)
```

Comment:
`TypeError` would be consistent with what Python would use if you simply omitted `**kwargs` (which would probably be a better idea, eventually):
```suggestion
    raise TypeError(msg)
```